ApplicationLibCode/ProjectDataModel/Summary/RimSummaryPlot.cpp

```diff
@@ -2884,16 +2884,45 @@ void RimSummaryPlot::assignPlotAxis( RimSummaryCurve* destinationCurve )
         ALL_TO_LEFT,
         ALL_TO_RIGHT,
         ALTERNATING,
-        USE_MATCHING_UNIT
+        USE_MATCHING_UNIT,
+        USE_MATCHING_VECTOR
     };
 
-    RiaDefines::PlotAxis plotAxis = RiaDefines::PlotAxis::PLOT_AXIS_LEFT;
-
     auto strategy = AxisAssignmentStrategy::USE_MATCHING_UNIT;
-    if ( strategy == AxisAssignmentStrategy::USE_MATCHING_UNIT )
+
+    auto destinationUnit = destinationCurve->unitNameY();
+    if ( destinationUnit.empty() ) strategy = AxisAssignmentStrategy::USE_MATCHING_VECTOR;
+
     {
-        auto destinationUnit = destinationCurve->unitNameY();
+        bool anyCurveWithUnit = false;
+
+        for ( auto c : summaryCurves() )
+        {
+            if ( c == destinationCurve ) continue;
+
+            if ( !c->unitNameY().empty() ) anyCurveWithUnit = true;
+        }
 
+        if ( !anyCurveWithUnit ) strategy = AxisAssignmentStrategy::USE_MATCHING_VECTOR;
+    }
```

Comment:
Maybe create a lambda for checking if there are curves without units? Introducing an extra scope level usually indicates that you should have made a function/lambda.